server/app.py

```diff
@@ -117,6 +118,11 @@ def server_error_handler(error):
 
     app.sentry = sentry
     sentry.init_app(app)
+
+    # instantiate registered provider classes (leave non-classes intact)
+    for key, provider in providers.items():
+        providers[key] = provider() if isinstance(provider, type) else provider
```

Comment:
The initialization is needed also on celery because on update_provider(celery task) you access the list of providers.
Better to create a method on superdesk.io, and the method returns the providers by name. In the method if provider is not created, you can create it.